### PR TITLE
improve manifest parsing error for connection constraints

### PR DESCRIPTION
**Pull request by mmandlis**
https://github.com/PolymerLabs/arcs/issues/681 and https://github.com/PolymerLabs/arcs/issues/684

### CODE REVIEW

File: runtime/manifest.js [hunk 1]
```diff
@@ -367,8 +367,26 @@ ${e.message}
     for (let connection of items.connections) {
       let fromParticle = manifest.findParticleByName(connection.from.particle);
       let toParticle = manifest.findParticleByName(connection.to.particle);
-      assert(fromParticle, `could not find particle ${fromParticle}`);
-      assert(toParticle, `could not find particle ${toParticle}`);
+      if (!fromParticle) {
+        let error = new Error(`could not find particle '${connection.from.particle}'`);
```
Comment (smalls) [inline runtime/manifest.js]: It looks to be old but I dig the literate `connection.from.particle` style code.

File: runtime/test/planner-tests.js [hunk 2]
```diff
@@ -170,7 +173,7 @@ describe('ConvertConstraintsToConnections', async () => {
     let {result, score} = results[0];
     assert.deepEqual(result.toString(),
 `recipe
-  create as view0
+  create as view0 // S
```
Comment (smalls) [inline runtime/test/planner-tests.js]: What's the `// S`?

Comment (smalls) [reply runtime/test/planner-tests.js]: (more instances below).

File: runtime/manifest.js (hunk 1 above)
Comment (mmandlis) [reply runtime/manifest.js]: the previous error wasn't working, as the fromParticle would be undefined when this error happens.

File: runtime/test/planner-tests.js (hunk 2 above)
Comment (mmandlis) [reply runtime/test/planner-tests.js]: `// S` is the type (like in real recipes that would be, for example: // Product) - see `schema S` above.
I'm debating whether i should rename it to `schema Thing` for example, but then it won't be consistent with the one letter particle names, like `particle A`.

Comment (smalls) [reply runtime/test/planner-tests.js]: Ah, I get it. I think (although I don't quite remember) I thought this was supposed to be `#S`, so more detail in the comment might help.

Comment (mmandlis) [reply runtime/test/planner-tests.js]: It was # S when we used # (and not //) for comments.

**Pull request merged**